### ISSUE TITLE
flutter build ios on macOS 11.2.3 fails to install BoringSSL-GRPC (0.0.7)

### ISSUE THREAD

**Issue by steveroe317** [firebase/firebase-ios-sdk#7881]
After creating a new flutter project and adding the dependencies from https://firebase.flutter.dev/docs/firestore/overview/

```
  firebase_core: "^0.7.0"
  cloud_firestore: "^0.16.0+1"
```

"flutter build ios" fails with 
```
    -> Installing BoringSSL-GRPC (0.0.7)
     > Git download
     > Git download
         $ /usr/bin/git clone https://github.com/google/boringssl.git
         /var/folders/6m/lz_6dkhn4t515t3pm9dmv27c0000gn/T/d20210411-8506-k9r3ta --template=

    [!] Error installing BoringSSL-GRPC
    [!] /usr/bin/git clone https://github.com/google/boringssl.git
    /var/folders/6m/lz_6dkhn4t515t3pm9dmv27c0000gn/T/d20210411-8506-k9r3ta --template=

    Cloning into '/var/folders/6m/lz_6dkhn4t515t3pm9dmv27c0000gn/T/d20210411-8506-k9r3ta'...
    fatal: the remote end hung up unexpectedly
```
Running `pod install` in the project's ios directory fails with a similar error.

However, running a git clone command directly succeeds
```
% /usr/bin/git clone https://github.com/google/boringssl.git /var/folders/6m/lz_6dkhn4t515t3pm9dmv27c0000gn/T/d20210411-8965-1ba66q3 --template=
Cloning into '/var/folders/6m/lz_6dkhn4t515t3pm9dmv27c0000gn/T/d20210411-8965-1ba66q3'...
remote: Enumerating objects: 725, done.
remote: Counting objects: 100% (725/725), done.
remote: Compressing objects: 100% (450/450), done.
remote: Total 80422 (delta 462), reused 450 (delta 261), pack-reused 79697
Receiving objects: 100% (80422/80422), 165.74 MiB | 31.63 MiB/s, done.
Resolving deltas: 100% (57071/57071), done.
Updating files: 100% (5497/5497), done.
```

How can I get flutter build ios to work with firebase dependencies?

The environment is:

macOS Big Sur 11.2.3
Flutter 2.0.4 • channel stable • https://github.com/flutter/flutter.git
Framework • revision b1395592de (10 days ago) • 2021-04-01 14:25:01 -0700
Engine • revision 2dce47073a
Tools • Dart 2.12.2
pod --version
1.10.1

[flutter_build_ios.output.txt](https://github.com/firebase/firebase-ios-sdk/files/6292180/flutter_build_ios.output.txt)
[pubspec copy.yaml.txt](https://github.com/firebase/firebase-ios-sdk/files/6292188/pubspec.copy.yaml.txt)
[Podfile.txt](https://github.com/firebase/firebase-ios-sdk/files/6292184/Podfile.txt)
[Podfile.lock.txt](https://github.com/firebase/firebase-ios-sdk/files/6292185/Podfile.lock.txt)

**Comment by paulb777** [comment]
This is likely a transitory networking issue. Please try again. If that doesn't work, you may have a corrupted `~/.cocoapods` directory

**Comment by steveroe317** [comment]
Just tried again, with the same results.  The error has been persistent over the last day or so.

Then ran `sudo rm -rf ~/.cocoapods/repos` and retried `flutter build ios` but got the same error.

**Comment by steveroe317** [comment]
What puzzles me is why `flutter build ios` and `pod install` fail, but the git clone command reported in the error messages succeeds.

**Comment by steveroe317** [comment]
It was a network issue.

I got past the error by disconnecting the wired connection to my Google wifi hub and running the commands over the wifi interface. To the same hub.

Some of the posted solutions for this problem involved changing git buffer size limits, which I tried without success.  Maybe the docking station (CalDigit USB Type C Pro Dock) has its own limits on buffer sizes?

After the build over the wifi interface succeeded, builds using the wired interface now work too.  So, there is a workaround.  It would be nice to know exactly why the wired interface failed to load BoringSSL-GRPC, but that's a separate issue.
